src/components/ActivityFeed/ActivityFeedItem.js

```diff
@@ -0,0 +1,530 @@
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import get from 'lodash.get';
+import { withTheme } from 'styled-components/native';
+
+// utils
+import { getThemeColors } from 'utils/themes';
+import { addressesEqual, getAssetData, getAssetsAsList } from 'utils/assets';
+import { createAlert } from 'utils/alerts';
+import { findMatchingContact } from 'utils/contacts';
+
+// components
+import {
+  formatAmount,
+  formatUnits,
+} from 'utils/common';
+import ListItemWithImage from 'components/ListItem/ListItemWithImage';
+import TankAssetBalance from 'components/TankAssetBalance';
+
+// constants
+import {
+  TYPE_RECEIVED,
+  TYPE_ACCEPTED,
+  TYPE_REJECTED,
+  TYPE_SENT,
+} from 'constants/invitationsConstants';
+import { COLLECTIBLE_TRANSACTION } from 'constants/collectiblesConstants';
+import {
+  TRANSACTION_EVENT,
+  TX_PENDING_STATUS,
+} from 'constants/historyConstants';
+import {
+  PAYMENT_NETWORK_ACCOUNT_DEPLOYMENT,
+  PAYMENT_NETWORK_ACCOUNT_TOPUP,
+  PAYMENT_NETWORK_ACCOUNT_WITHDRAWAL,
+  PAYMENT_NETWORK_TX_SETTLEMENT,
+} from 'constants/paymentNetworkConstants';
+import { ACCOUNT_TYPES } from 'constants/accountsConstants';
+import { USER_EVENT, PPN_INIT_EVENT, WALLET_CREATE_EVENT, WALLET_BACKUP_EVENT } from 'constants/userEventsConstants';
+import { BADGE_REWARD_EVENT } from 'constants/badgesConstants';
+import { SET_SMART_WALLET_ACCOUNT_ENS } from 'constants/smartWalletConstants';
+import { SettlementItem } from 'components/ActivityFeed/SettlementItem';
+
+// selectors
+import { activeAccountAddressSelector, supportedAssetsSelector, bitcoinAddressSelector } from 'selectors';
+import { accountAssetsSelector } from 'selectors/assets';
+import { activeBlockchainSelector } from 'selectors/selectors';
+
+// types
+import type { Asset } from 'models/Asset';
+import type { ContactSmartAddressData, ApiUser } from 'models/Contacts';
+import type { Theme } from 'models/Theme';
+import type { RootReducerState } from 'reducers/rootReducer';
+import type { EnsRegistry } from 'reducers/ensRegistryReducer';
+
+type Props = {
+  type?: string,
+  asset?: string,
+  isPending?: boolean,
+  supportedAssets: Asset[],
+  selectEvent: Function,
+  contacts: ApiUser[],
+  contactsSmartAddresses: ContactSmartAddressData[],
+  ensRegistry: EnsRegistry,
+  theme: Theme,
+  event: Object,
```

Comment:
Yeah, it would be nice, but we have different types of events (the transaction event is very different than for example user event). Also we'd need to refactor the types all over the place which is out of scope of this PR.